### DISCUSSION TITLE
Having trouble reinstalling toltec

### DISCUSSION THREAD

**Discussion by cindy-jth** [toltec-dev/toltec#469] [Support] [answered]
![image](https://user-images.githubusercontent.com/65687259/137604878-ebff53b4-931a-4a44-ab17-fbadbf14895e.png)

**Comment by LinusCDE** [comment, marked as answer]
Hi,

**EDIT**: Before just saw the first error. Sorry. Try to do a `rm bootstrap` and then rerun the commands from the readme first. If that works out, just ignore the rest of this answer, since you might not need to remove the existing installation (Evildos install script doesn't support firmware updates out-of-the-box but has a reenable script for that).

---

seems like you didn't uninstall toltec the last time (`Folder /home/root/.entware exists!`). I would take the wild guess here that your device was update the last time and hence doesn't have any residual service file which would result in an unclean uninstall. In your case, doing a `rm -r /home/root/.entware` should do the trick (but please check whether you can find the file `/lib/systemd/system/opt.mount` first. In case you do, a proper uninstallation is needed instead).

You might also want to save some of the files in `/home/root/.entware/etc`. Some applications save configs there that could get deleted.

Also please not that we are not strictly compatible with Evildo's entware installer anymore. You will only have access to the entware repo and not the toltec repo (which includes most rm specific apps). And adding that repo there after the fact is not recommended since the toltec bootstrap/toltecctl does some specific things that entware doesn't by default.

**TL;DR**
 - If the file `/lib/systemd/system/opt.mount` does not exist and you've backed up configs you want from `/home/root/.entware/etc`, delete the .entware folder (= /opt) by running `rm -r /home/root/.entware`).
 - After that download and run the bootstrap using the commands in the readme (main site of this repo).

**Reply by Eeems** [reply]
@Cindy-Jie-01 was this helpful?
